addons/hr_holidays/models/hr_leave.py

```diff
@@ -1166,22 +1166,22 @@ def _check_approval_update(self, state):
             return
 
         current_employee = self.env.user.employee_id
-        is_officer = self.env.user.has_group('hr_holidays.group_hr_holidays_user')
         is_manager = self.env.user.has_group('hr_holidays.group_hr_holidays_manager')
-
+        is_officer = self.env.user.has_group('hr_holidays.group_hr_holidays_user')
         for holiday in self:
+            is_approver = holiday.employee_id.leave_manager_id == self.env.user
```

Comment:
this variable seems to not be used in every iteration I think it would be better to put it directly where it's needed (l1190)

Reply:
I wanted to make the conditions expressive like `is_manager` and `is_officer` as I needed to read the condition twice first time I see it and it was not clear what it is checking